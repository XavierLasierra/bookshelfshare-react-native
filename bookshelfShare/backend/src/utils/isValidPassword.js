const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
