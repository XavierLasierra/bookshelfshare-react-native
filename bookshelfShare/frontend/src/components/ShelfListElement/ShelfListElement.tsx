import React from 'react';
import {
  View, Text, TouchableOpacity, Image
} from 'react-native';
import ShelfIcon from '../../assets/shelfIcon.svg';

import styles from './shelfListElement.styles';

export default function ShelfListElement({ shelf, loggedUserId }: any) {
  const filteredUserList = shelf.users.filter(({ _id }: any) => _id !== loggedUserId);

  return (
    <TouchableOpacity style={styles.shelfContainer}>
      <ShelfIcon width={35} height={35} />
      <View style={styles.topContainer}>
        <Text style={styles.shelfName}>{shelf.name}</Text>

      </View>
      <View>
        <View style={styles.photosContainer}>
          <Image
            source={{ uri: filteredUserList[0]?.photo }}
            style={styles.sharedUserPhoto}
          />
          {filteredUserList.length > 2 && (
          <Text style={styles.sharedUserNumber}>
            {' '}
            +
              {filteredUserList.length - 1 }
          </Text>
          )}
        </View>
        <Text style={styles.bookCounter}>
          {shelf.books.length}
          {' '}
          {shelf.books.length === 1 ? 'book' : 'books'}
        </Text>

      </View>
    </TouchableOpacity>
  );
}
