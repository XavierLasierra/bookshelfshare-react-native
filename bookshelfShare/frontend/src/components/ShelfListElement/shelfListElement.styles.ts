import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  shelfContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: stylesConstants.colors.detail,
    height: 85,
    borderRadius: 10,
    marginVertical: 10,
    paddingVertical: stylesConstants.padding.small,
    paddingHorizontal: stylesConstants.padding.mid
  },
  photosContainer: {
    flexDirection: 'row-reverse',
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  sharedUserPhoto: {
    width: 30,
    height: 30,
    borderRadius: 100
  },
  sharedUserNumber: {
    fontSize: stylesConstants.fontSize.mid,
    marginRight: 5
  },
  shelfName: {
    fontSize: stylesConstants.fontSize.mid,
    letterSpacing: 1,
    width: 150,
    fontWeight: 'bold'
  },
  bookCounter: {
    fontSize: stylesConstants.fontSize.small,
    textAlign: 'right',
    width: 70
  },
  photoCounterContainer: {
    flexDirection: 'column-reverse',
    height: '100%',
    justifyContent: 'space-between',
    alignItems: 'flex-end'
  }
});
