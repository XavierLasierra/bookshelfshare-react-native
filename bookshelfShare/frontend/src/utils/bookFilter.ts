export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
