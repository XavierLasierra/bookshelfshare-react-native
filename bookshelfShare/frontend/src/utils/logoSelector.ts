import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
