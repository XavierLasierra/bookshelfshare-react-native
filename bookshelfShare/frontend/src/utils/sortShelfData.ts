export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
