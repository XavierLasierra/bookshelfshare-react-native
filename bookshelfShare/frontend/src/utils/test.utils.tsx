import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
