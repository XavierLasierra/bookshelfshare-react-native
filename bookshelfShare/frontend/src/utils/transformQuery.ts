interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
