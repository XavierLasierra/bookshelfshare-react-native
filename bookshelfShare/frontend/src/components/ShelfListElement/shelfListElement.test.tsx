import React from 'react';
import ShelfListElement from './ShelfListElement';
import { render, fireEvent } from '../../utils/test.utils';
import shelfMock from '../../mocks/shelf.mock';
import { loadCurrentShelf } from '../../redux/actions/currentShelf.creator';

jest.mock('../../redux/actions/currentShelf.creator');

const navigation = {
  push: jest.fn()
};

describe('Given a ShelfListElement component', () => {
  describe('When it is rendered', () => {
    describe('And the shelf has 1 book', () => {
      let screen: any;
      beforeEach(() => {
        (loadCurrentShelf as jest.Mock).mockReturnValue({ type: '' });
        screen = render(<ShelfListElement
          shelf={{ ...shelfMock, books: ['book'] }}
          navigation={navigation}
        />);
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });

      describe('And you click in shelfDetailPageButton', () => {
        beforeEach(() => {
          const shelfDetailPageButton = screen.getByTestId('shelfDetailPageButton');
          fireEvent.press(shelfDetailPageButton);
        });
        test('Then loadCurrentShelf should have been called', () => {
          expect(loadCurrentShelf).toHaveBeenCalled();
        });
        test('Then navigation.push should have been called', () => {
          expect(navigation.push).toHaveBeenCalled();
        });
      });
    });
    describe('And the shelf has 0 books', () => {
      test('Then should match the snapshot', () => {
        const screen = render(<ShelfListElement shelf={{ ...shelfMock, books: [] }} />);
        expect(screen).toMatchSnapshot();
      });
    });
  });
});
