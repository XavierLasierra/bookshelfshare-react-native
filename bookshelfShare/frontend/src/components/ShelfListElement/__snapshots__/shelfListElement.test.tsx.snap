// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ShelfListElement component When it is rendered And the shelf has 0 books Then should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9CB83",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
  testID="shelfDetailPageButton"
>
  <IconMock
    height={35}
    width={35}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "letterSpacing": 1,
        "width": 150,
      }
    }
  >
    Shelf
  </Text>
  <View
    style={
      Object {
        "height": 65,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row-reverse",
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "photo",
          }
        }
        style={
          Object {
            "borderRadius": 100,
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginRight": 5,
          }
        }
      >
         
        +
        3
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlign": "right",
        }
      }
    >
      0
       
      books
    </Text>
  </View>
</View>
`;

exports[`Given a ShelfListElement component When it is rendered And the shelf has 1 book Then should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9CB83",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
  testID="shelfDetailPageButton"
>
  <IconMock
    height={35}
    width={35}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "letterSpacing": 1,
        "width": 150,
      }
    }
  >
    Shelf
  </Text>
  <View
    style={
      Object {
        "height": 65,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row-reverse",
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "photo",
          }
        }
        style={
          Object {
            "borderRadius": 100,
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginRight": 5,
          }
        }
      >
         
        +
        3
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlign": "right",
        }
      }
    >
      1
       
      book
    </Text>
  </View>
</View>
`;
